core/src/main/java/com/novoda/merlin/registerable/Registrar.java

```diff
@@ -61,8 +66,16 @@ public void registerBindable(Bindable bindable) {
     }
 
     public void clearRegistrations() {
-        connectables.clear();
-        disconnectables.clear();
-        bindables.clear();
+        if (connectables != null) {
+            connectables.clear();
+        }
```

Comment:
Bit shocked that these were never picked up before 😬  especially given we throw when the corresponding method is not called when registering. Should always try to use the annotation where possible!